distsql: set NodeID and FlowID in logging contexts

RaduBerinde opened the pull request:
Please ignore the first commit (r1), it is a separate PR.

Sample log output:

```
I160727 14:38:38.212306 sql/distsql/tablereader.go:233  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] [TableReader: 51] pushing row [83 11]
I160727 14:38:38.212335 sql/distsql/tablereader.go:233  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] [TableReader: 51] pushing row [92 11]
I160727 14:38:38.212345 sql/distsql/outbox.go:101  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] flushing outbox
I160727 14:38:38.212364 sql/distsql/outbox.go:114  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] outbox flushed
I160727 14:38:38.212377 sql/distsql/inbound.go:80  [node: 3] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] inbound stream pushing row [81 9]
```

@tschottdorf @jordanlewis

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/8066)

<!-- Reviewable:end -->

RaduBerinde committed:
distsql: set NodeID and FlowID in logging contexts

Also fixing a nil session context log panic (with verbosity).

Sample log output:

```
I160803 15:14:54.175240 sql/distsql/tablereader.go:233  [node=2,flow=d9a2137f,TableReader=51] pushing row [46 10]
I160803 15:14:54.175259 sql/distsql/tablereader.go:233  [node=2,flow=d9a2137f,TableReader=51] pushing row [55 10]
I160803 15:14:54.175260 sql/distsql/inbound.go:81  [node=3,flow=d9a2137f] inbound stream pushing row [8 8]
I160803 15:14:54.175268 sql/distsql/inbound.go:81  [node=3,flow=d9a2137f] inbound stream pushing row [17 8]
```

jordanlewis commented:
Reviewed 22 of 22 files at r1.
Review status: 16 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhgf7sWld9H2We5tUy:1374822074) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
> ```Go
> 	simpleFlowConsumer RowReceiver,
> ) *Flow {
> 	// TODO(radu): UUIDs are pretty long, perhaps an int64 would be sufficient
> ```

Agreed that UUIDs are a bit long, especially for outputting in the logs. Maybe we could print a short prefix instead if we'd like to keep the UUID internally.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: 16 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhhHglUfjox6Vy1Hp_:587908693) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> Agreed that UUIDs are a bit long, especially for outputting in the logs. Maybe we could print a short prefix instead if we'd like to keep the UUID internally.
</details>
`util/uuid.UUID.Short()`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Current diff r1->r3.

---

Review status: 16 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhl-apZogans79g1Xh:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> `util/uuid.UUID.Short()`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KNhl03sEfeMgxi0zspP:-29048018)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 22 of 22 files at r1, 5 of 6 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhpxSftExPWMjAvSUb:2119796242) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> Done.
</details>
probably good to remove this TODO now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Current diff r1->r4.

---

Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhyi8QFLVTABj3pW8Y:1597093114) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> probably good to remove this TODO now.
</details>
Ah, duh, thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KNhypA-LFIJpu6Id4IY:-29047987)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Updated. I recommend looking at the diff in github as Reviewable is thoroughly confused. @andreimatei 

---

Review status: 3 of 8 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOGqPW24Y84FNRr-hkN:43079080)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
Could we get the context info in front of the file/line num in the logs?

RaduBerinde commented:
It would require a bit of dancing (we trace the message without the file/line) but I don't see why not. @petermattis what do you think?

Either way it would be a separate PR, this PR isn't related to the general format of log messages.

---

Review status: 3 of 8 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOH2UELPn4AVmnx4Sf7:-390243071)*
<!-- Sent from Reviewable.io -->

andreimatei commented on sql/distsql/joinreader.go:
```diff
@@ -95,10 +93,10 @@ func (jr *joinReader) mainLoop() error {
 	spans := make(sqlbase.Spans, 0, joinReaderBatchSize)
 
-	// TODO(radu): add info about the joinreader in the context.
+	ctx := log.WithLogTagInt(jr.ctx, "JoinReader", int(jr.desc.ID))
```
can this move to the ctor?

andreimatei commented on server/server.go:
```diff
@@ -421,6 +420,7 @@ func (s *Server) Start() error {
 	s.node.startWriteSummaries(s.ctx.MetricsSampleInterval)
 
 	s.sqlExecutor.SetNodeID(s.node.Descriptor.NodeID)
```
I forgot how this works; when does is this `s.node.NodeID` assigned? This seems like a late place to init these fields on the executor and distsqlServer.

andreimatei commented:
LGTM

RaduBerinde commented:
TFTR!

---

Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[server/server.go, line 422 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhg66J0R41iRBEILdG-r5-422:-KOH5Vm_gVJtrqgzRLgK:-1772191250) ([raw file](https://github.com/cockroachdb/cockroach/blob/c07b5c3cc95d142f71f23b1857a35afbca4e14b0/server/server.go#L422)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> I forgot how this works; when does is this `s.node.NodeID` assigned? This seems like a late place to init these fields on the executor and distsqlServer.
</details>
Unfortunately it can be as late as the `node.start()` call above (`start->initStores->initNodeID`).

---

*[sql/distsql/joinreader.go, line 96 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhg66J0R41iRBEILdN-r5-96:-KOH5jGSGKTdd1sYEadF:1972804061) ([raw file](https://github.com/cockroachdb/cockroach/blob/c07b5c3cc95d142f71f23b1857a35afbca4e14b0/sql/distsql/joinreader.go#L96)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> can this move to the ctor?
</details>
Done. Also renamed some `ctx *FlowCtx`s to `flowCtx`, and changed the embedded `Context` in `FlowCtx` and `ServerContext` to regular members to avoid confusion.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOH7myiRpBaC5ABsTXY:-485616692)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
LGTM

On Wed, Aug 3, 2016 at 4:45 PM, RaduBerinde <notifications@github.com>
wrote:

> TFTR!
> ------------------------------
>
> Review status: 0 of 9 files reviewed at latest revision, 3 unresolved
> discussions, some commit checks pending.
> ------------------------------
>
> *server/server.go, line 422 [r5]
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhg66J0R41iRBEILdG-r5-422:-KOH5Vm_gVJtrqgzRLgK:-1772191250>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/c07b5c3cc95d142f71f23b1857a35afbca4e14b0/server/server.go#L422>):*
> *Previously, andreimatei (Andrei Matei) wrote…*
>
> I forgot how this works; when does is this s.node.NodeID assigned? This
> seems like a late place to init these fields on the executor and
> distsqlServer.
>
>
> Unfortunately it can be as late as the node.start() call above (
> start->initStores->initNodeID).
>
> ------------------------------
>
> *sql/distsql/joinreader.go, line 96 [r5]
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhg66J0R41iRBEILdN-r5-96:-KOH5jGSGKTdd1sYEadF:1972804061>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/c07b5c3cc95d142f71f23b1857a35afbca4e14b0/sql/distsql/joinreader.go#L96>):*
> *Previously, andreimatei (Andrei Matei) wrote…*
>
> can this move to the ctor?
>
>
> Done. Also renamed some ctx *FlowCtxs to flowCtx, and changed the
> embedded Context in FlowCtx and ServerContext to regular members to avoid
> confusion.
>
> ------------------------------
>
> *Comments from Reviewable
> <https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOH7myiRpBaC5ABsTXY:-485616692>*
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/8066#issuecomment-237367433>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAXBcV0ZAEfu8IRgn1JCbjZfGKPfNOtjks5qcP3cgaJpZM4JWgWU>
> .
>

petermattis commented:
Putting the context tags before the file/line number is doable, though mildly irritating. I don't see a huge benefit to doing so. More interesting would be to include the goroutine ID. See https://github.com/petermattis/goid.

---

Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOHE-YeKk7MhlQiL1GF:-9597875)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
I like the `goid` stuff! I know that the go folks are dead set against providing an API to get the current goroutine; are we reasonably confident we will always be able to have a working hack for future go versions?

---

Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOHJqRTKSxe1sEreUPN:208303033)*
<!-- Sent from Reviewable.io -->
